app/javascript/mastodon/features/audio/index.js

```diff
@@ -202,7 +203,7 @@ class Audio extends React.PureComponent {
     const muted = !this.state.muted;
 
     this.setState({ muted }, () => {
-      this.audio.muted = muted;
+      this.gainNode.gain.value = muted ? 0 : this.state.volume;
```

Comment:
Sorry I missed that, but it needs to check that this.gainNode is defined, which might not be the case if the user attempts to change the volume before starting playback.

```suggestion
      if (this.gainNode) {
        this.gainNode.gain.value = muted ? 0 : this.state.volume;
      }
```